viewing_party/party.py

```diff
@@ -1,23 +1,184 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    movie_dict = {}
+
+    if not title or not genre or not rating:
+        return None
+    else:
+        movie_dict['title'] = title
+        movie_dict['genre'] = genre
+        movie_dict['rating'] = rating
+    
+    return movie_dict
+
+def add_to_watched(user_data, movie):
+
+    user_data["watched"].append(movie)
+    
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+
+    user_data["watchlist"].append(movie)
+    
+    return user_data
+
+def watch_movie(user_data, title):
+
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+            break
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+
+    if not user_data["watched"]:
+        return 0.0
+    
+    total_rating = 0
+    for movie in user_data["watched"]:
+        total_rating += movie["rating"] 
+            
+    average = total_rating / len(user_data["watched"])
+
+    return average
+
+def get_most_watched_genre(user_data):
+    genre_count = {"Fantasy": 0, "Intrigue": 0, "Action": 0} #change method so that other genres can be included
+
+    if not user_data["watched"]:
+        return None 
+    
+    for movie in user_data["watched"]:
+        genre = movie["genre"]
+        if genre in genre_count:
+            genre_count[genre] += 1   
+    top_genre = max(genre_count, key=genre_count.get)
+
+    return top_genre

```

Comment:
So right now, your code is dependent on if the movie has a genre of one the three listed on `line 54`. Though it passes the tests, as coders we want to think beyond the tests as well (what is a movie had a genre of romance?). How could we implement a version of this that is agnostic to specific genres? (Hint: Hash Maps/Frequency Maps)